src/protocol/oprf/shuffle/mod.rs

```diff
@@ -0,0 +1,403 @@
+use std::ops::{Add, AddAssign};
```

Comment:
I think that Daniel pointed this out, but this has nothing to do with the OPRF.  It should replace our existing shuffle.

Reply:
I agree that this should be the shuffle that we use in both approaches. Alex and I discussed this. Alex seemed to prefer that we still put it in an OPRF related module even when it is a basic protocol (this is also the case for operations on Boolean shares). The main reason is that we have a more clear separation between old and new components in case we want to migrate away from the old IPA.

Reply:
I don't think shuffle implementations are at the state where we can switch between them. The new one requires special input format (moving from struct { secret_share1, secret_share2, ... } to secret_share { struct1, struct2 }) while the old one requires them to be reshareable only. It would be ideal if shuffle is generalized over some sort of a trait, but I don't think it is atm.

So we may want to keep OPRF-specific things inside a different module

Reply:
This isn't OPRF-specific, it's just a different shuffle.  It's OK if we have two shuffles for now, but this shouldn't have to be bundled in with the OPRF.